mpiu.c

```diff
@@ -0,0 +1,460 @@
+#include <stddef.h>
```

Comment:
I put memory allocator, gather/scatter, alltoall, and segmenter to the MPIU namespace.